server/protocols/jmap/src/main/java/org/apache/james/jmap/Version.java

```diff
@@ -0,0 +1,50 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public enum Version {
+    DRAFT("draft"),
+    RFC8621("rfc-8621");
+
+    public static Version of(Optional<String> maybeVersion) {
```

Comment:
IMO taking an Optional as a parameter is a code smell.

We take business decisions from within a pojo (rejecting illegal versions, having draft as a default).

I would prefer having this handling done in JmapServer.